### PR TITLE
[Component/Core] Remove deprecated code

### PR DESCRIPTION
**Pull request by Wojdylak**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 2.0
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | -
| License         | MIT

#15376

### CODE REVIEW

File: src/Sylius/Component/Core/Model/OrderInterface.php [hunk 1]
```diff
@@ -120,10 +120,6 @@ public function setCustomerWithAuthorization(?BaseCustomerInterface $customer):
 
     public function isCreatedByGuest(): bool;
 
-    public function getCreatedByGuest(): bool;
-
-    public function setCreatedByGuest(bool $createdByGuest): void;
-
```
Comment (Rafikooo) [inline src/Sylius/Component/Core/Model/OrderInterface.php:126]: I know this removal is obvious, due to the deprecation of methods in the implementing class, but I still wonder if we should deprecate these interface methods first in Sylius 1.13

File: src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php [hunk 2]
```diff
@@ -56,15 +56,6 @@ public function createListQueryBuilder(): QueryBuilder
...
 
-    public function createSearchListQueryBuilder(): QueryBuilder
-    {
-        return $this->createListQueryBuilder()
-            ->leftJoin('o.items', 'item')
-            ->leftJoin('item.variant', 'variant')
-            ->leftJoin('variant.product', 'product')
-        ;
-    }
-
```
Comment (Rafikooo) [inline src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php:67]: Same case as my previous comment

File: src/Sylius/Component/Core/Taxation/Applicator/OrderItemsTaxesApplicator.php [hunk 3]
```diff
@@ -86,20 +72,20 @@ public function apply(OrderInterface $order, ZoneInterface $zone): void
...
+//            $taxRate = $this->taxRateResolver->resolve($item->getVariant(), ['zone' => $zone]);
+//            if (null === $taxRate) {
+//                continue;
+//            }
+//
+//            $totalTaxAmount = $this->calculator->calculate($item->getTotal(), $taxRate);
+//
+//            $this->distributeTaxesToUnits($totalTaxAmount, $item->getQuantity(), $item, $taxRate);
+//        }
+//    }
```
Comment (Rafikooo) [inline src/Sylius/Component/Core/Taxation/Applicator/OrderItemsTaxesApplicator.php]: Some missed leftovers

**Review by Rafikooo — DISMISSED** [review]

File: src/Sylius/Component/Core/Model/OrderInterface.php (hunk 1 above)
Comment (Wojdylak) [reply src/Sylius/Component/Core/Model/OrderInterface.php:126]: https://github.com/Sylius/Sylius/pull/15376

File: src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php (hunk 2 above)
Comment (Wojdylak) [reply src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php:67]: https://github.com/Sylius/Sylius/pull/15376

**Review by TheMilek — APPROVED** [review]

**Review by GSadee — APPROVED** [review]

**Pull request merged**